src/umdsst/Geometry/Geometry.cc

```diff
@@ -37,13 +64,71 @@ namespace umdsst {
       atlas::option::halo(0)));
 
     atlasFieldSet_.reset(new atlas::FieldSet());
-    atlasFieldSet_->add(atlasFunctionSpace_->lonlat());
+//  atlasFieldSet_->add(atlasFunctionSpace_->lonlat());
+    for (int i = 0; i < other.atlasFieldSet()->size(); i++) {
+      atlasFieldSet_->add((*other.atlasFieldSet())[i]);
+    }
   }
 
 // ----------------------------------------------------------------------------
 
   Geometry::~Geometry() {}
 
+// ----------------------------------------------------------------------------
+
+  void Geometry::loadLandMask(const eckit::Configuration &conf) {
+    // use an globalLandMask to read the data on root PE only.
+    atlas::Field globalLandMask = atlasFunctionSpace_->createField<int>(
+                                  atlas::option::levels(1) |
+                                  atlas::option::name("gmask") |
+                                  atlas::option::global());
+    auto fd = make_view<int, 2>(globalLandMask);
+
+    if (globalLandMask.size() != 0) {
+      int lat = 0, lon = 0;
+      std::string filename;
+
+      if (!conf.get("landmask.filename", filename))
+        util::abor1_cpp("Geometry::loadLandMask(), Get filename failed.",
+          __FILE__, __LINE__);
+      else
+        oops::Log::info() << "In Geometry::loadLandMask(), filename = "
+                          << filename << std::endl;
+
+      // Open netCDF file
+      netCDF::NcFile file(filename.c_str(), netCDF::NcFile::read);
+      if (file.isNull())
+        util::abor1_cpp("Geometry::loadLandMask(), Create netCDF file failed.",
+          __FILE__, __LINE__);
+
+      // get file dimensions
+      lat = static_cast<int>(file.getDim("lat").getSize());
+      lon = static_cast<int>(file.getDim("lon").getSize());
+
+      // get landmask data
+      netCDF::NcVar varLandMask;
+      varLandMask = file.getVar("landmask");
+      if (varLandMask.isNull())
+        util::abor1_cpp("Get var landmask failed.", __FILE__, __LINE__);
+
+      int dataLandMask[lat][lon];
+      varLandMask.getVar(dataLandMask);
+
+      int idx = 0;
+      for (int j = 0; j < 180; j++)
```

Comment:
Can you remove the hardcoded 180 and 360 here and use the actual dimensions?

Reply:
Just updated, thanks.